src/components/MapboxMap.tsx

```diff
@@ -0,0 +1,179 @@
+import { useRef, useEffect } from 'react';
+import mapboxgl, { Map, Marker, MapboxOptions, LngLatBounds, MarkerOptions, LngLat } from 'mapbox-gl';
+import { Result, useSearchState } from '@yext/search-headless-react';
+import { useDebouncedFunction } from '../hooks/useDebouncedFunction';
+import ReactDOM from 'react-dom';
+
+/**
+ * A functional component that can be used to render a custom marker on the map.
+ *
+ * @public
+ */
+export type PinComponent<T> = (props: { result?: Result<T>, index?: number }) => JSX.Element;
+
+/**
+ * A function use to derive a result's coordinate.
+ *
+ * @public
+ */
+export type CoordinateGetter<T> = (result: Result<T>) => Coordinate | undefined;
+
+/**
+ * Coordinate use to represent the result's location on a map.
+ *
+ * @public
+ */
+export interface Coordinate {
+  /** The latitude of the location. */
+  latitude: number,
+  /** The longitude of the location. */
+  longitude: number
+}
+
+/**
+ * A function which is called when user drag the map.
+ *
+ * @public
+ */
+export type OnDragHandler = (center: LngLat, bounds: LngLatBounds) => void;
+
+/**
+ * Props for the {@link MapboxMap} component.
+ *
+ * @typeParam T - Type of "rawData" field of the results use in the map
```

Comment:
is `typeParam` a tag that API extractor recognizes? I think it might just ignore this and not add it to the docs

Reply:
I thought it does looking at their docs: https://api-extractor.com/pages/tsdoc/tag_typeparam/. But looking at our generated docs (seems like it never got implemented? https://github.com/microsoft/rushstack/issues/1635). I will just update this to be a normal comment then.